drivers/network/afd/afd/write.c

```diff
@@ -569,6 +569,8 @@ AfdPacketSocketWriteData(PDEVICE_OBJECT DeviceObject, PIRP Irp,
     PAFD_FCB FCB = FileObject->FsContext;
     PAFD_SEND_INFO_UDP SendReq;
     KPROCESSOR_MODE LockMode;
+    INT FullSendLen, LoopIdx; // accumulator for full packet length
+    char pktbuf[4096];  // local packet assembly buffer
```

Comment:
What justifies that 4096 is a good length? Why not 2048 or 65536 ?

Reply:
That is a valid question. I chose it based on remembering buffer sizes used in network code I have seen in the past which used it because other MAC layers supported it as max routable packet size(token ring I think). In truth it is more or less arbitrary because most UDP packets using it are small. I did consider 65536 (max allowable packet size in many places) or using 1550 (prevalent ethernet MTU size), but 2048 may be a better choice being the round base 2 size just above the ethernet MTU. As for hard coding, maybe a constant definition would be better ?   Make a suggestion with your preference and I'll accept it.

Reply:
The point of my comment is that there is a hardcoded size being used there, but then in the loop, you are copying (and appending) into that buffer data blocks, whose different sizes are known only at runtime. And so it can happen that crafted blocks could be sent with their sizes made in such a way that the copy operation would overflow that large _on-stack_ buffer and corrupt everything.